File: Android_Study/Example/DoitMission-07_FragmentChange1/app/src/main/java/com/satelite54/doitmission_07_fragmentchange/MainActivity.java
package com.satelite54.doitmission_07_fragmentchange;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    Button button;
    final int REQUEST_CODE_MENU = 101;
    final int MANAGEMENT1 = 1;
    final int MANAGEMENT2 = 2;
    final int MANAGEMENT3 = 3;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), MainActivity2.class);
                startActivityForResult(intent, REQUEST_CODE_MENU);
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        switch (resultCode) {
            case MANAGEMENT1: makeToast("고객관리 호출됨");
                break;
            case MANAGEMENT2: makeToast("매출 관리");
                break;
            case MANAGEMENT3: makeToast("상품 관리");
                break;
        }
    }

    public void makeToast(String str) {
        Toast.makeText(this, str, Toast.LENGTH_LONG).show();
    }
}

File: Android_Busan_it/MyActivityintent5/app/src/main/java/com/example/myactivityintent5/MainActivity.kt
package com.example.myactivityintent5

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.example.myactivityintent5.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    lateinit var v : ActivityMainBinding
    var operator : Char = '\u0000'
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        v = ActivityMainBinding.inflate(layoutInflater)
        var view = v.root
        setContentView(view)

        v.btnNewActivity.setOnClickListener {

            var intent = Intent(applicationContext, SecondActivity::class.java)
            intent.putExtra("num1", v.num1.text.toString().toInt())
            intent.putExtra("num2", v.num2.text.toString().toInt())
            intent.putExtra("operator", operator)
            startActivityForResult(intent, 0)
        }

        v.rGroup.setOnCheckedChangeListener { radioGroup, radiobtnId ->
            when(radiobtnId) {
               R.id.radio1 -> {
                   operator = '+'
               }
                R.id.radio2 -> {
                    operator = '-'
                }
                R.id.radio3 -> {
                    operator = '*'
                }
                R.id.radio4 -> {
                    operator = '/'
                }
            }
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        if(Activity.RESULT_OK == resultCode) {
            var hap = data!!.getIntExtra("Hap", 0)
            Toast.makeText(applicationContext, "결과 : ${hap}", Toast.LENGTH_SHORT).show()
        }
    }
}